File: main/jump-game/jump-game-slow.cs
public class Solution {
    public bool CanJump(int[] nums) {
        var istop = nums.Length;
        var reach = new bool[istop];
        reach[0] = true;

        for (var i = 0; i != istop; ++i) {
            if (reach[i]) {
                var jstop = Math.Min(istop, i + nums[i] + 1);
                for (var j = i + 1; j != jstop; ++j) reach[j] = true;
            }
        }

        return reach[istop - 1];
    }
}

File: main/jump-game/jump-game-slow.c
static inline int min(const int a, const int b) {
    return a < b ? a : b;
}

bool canJump(int *const nums, const int numsSize) {
    const int last = numsSize - 1;
    bool reach[numsSize];
    reach[0] = true;
    memset(reach + 1, 0, last * sizeof reach[1]);

    for (int i = 0; i != numsSize; ++i) {
        if (reach[i]) {
            for (int j = min(last, i + nums[i]); j != i; --j)
                reach[j] = true;
        }
    }

    return reach[last];
}
